src/Raven.Server/Documents/DocumentFlags.cs

```diff
@@ -9,24 +9,17 @@ public enum DocumentFlags
 
         Artificial = 0x1,
         Versioned = 0x2,
-        Reserved1 = 0x4,
-        Reserved2 = 0x8,
+        Revision = 0x4,
+        Reserved = 0x8,
 
         FromIndex = 0x10,
-        FromVersionStorage = 0x20,
+        Reserved1 = 0x20,
         FromReplication = 0x40,
-        Reserved3 = 0x80,
+        FromSmuggler = 0x80,
 
-        HasAttachments = 0x100
-    }
-
-    [Flags]
-    public enum NonPersistentDocumentFlags
```

Comment:
Personally I'm not a fan of having the ability to skip versioning of a single document. This requires transform script anyway so this can be filtered-out there if someone really needs it. Or revisions can be imported, and removed manually if the Studio UI will be good enough to handle that.

When it comes to saving the flags, of course every debug information can be useful, but in this case I would go and say that this would be confusing for the users. It's really hard to say if there is any values behind flag LegacyRevision for example. It was imported, it was saved, it is in new versioning storage, so someone might think what is the problem and why it is marked as a LegacyRevision, what are the implications and how can this be fixed.